### PR TITLE
[fix][authorization] Fix the return value of canConsumeAsync

### PR DESCRIPTION
**Pull request by nodece**
### Motivation

When an exception occurs in the `canConsumeAsync` method, we should throw that.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

**Review by lhotari — APPROVED** [review]

File: pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/PulsarAuthorizationProvider.java [hunk 1]
```diff
@@ -140,7 +140,7 @@ public CompletableFuture<Boolean> canConsumeAsync(TopicName topicName, String ro
                 }).exceptionally(ex -> {
                     log.warn("Client with Role - {} failed to get permissions for topic - {}. {}", role, topicName,
                             ex.getMessage());
-                    return null;
+                    throw FutureUtil.wrapToCompletionException(ex);
```
Comment (lhotari) [inline pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/PulsarAuthorizationProvider.java:143]: Is the logging just extra noise if the exception gets rethrown?  Could we remove the `.exceptionally` completely?

I guess `return null;` is wrong and could be `return false;` if the desire is to just log and continue by returning `false`.

Comment (michaeljmarshall) [reply pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/PulsarAuthorizationProvider.java:143]: We already log the "error" multiple times. We log it in this class as a warning. We also have these logs for handling the result of this method:

https://github.com/apache/pulsar/blob/952187279227fbf46af0f68d31ede1f5567e8075/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java#L888-L902

After this change, we could have 3 lines and a stack trace. I think that is too many. I would prefer to simplify the code path and only log a single failure/error/rejection once.

**Review by michaeljmarshall — CHANGES_REQUESTED** [review]
My preference is to remove the `exceptionally` block all together and to make sure that pulsar logs the errors accordingly. Since this `AuthorizationProvider` is pluggable, it simplifies the flow if implementations can rely on pulsar to log appropriately in my opinion.

Let me know what you think @nodece. Thanks!

**Comment by nodece** [comment]
I want to remove all `exceptionally`, what do you think of that?

**Review by coderzc — APPROVED** [review]

**Review by michaeljmarshall — APPROVED** [review]
LGTM, thanks @nodece!

**Pull request merged**